homeassistant/components/alexa/handlers.py

```diff
@@ -976,7 +976,12 @@ def temperature_from_object(hass, temp_obj, interval=False):
             service = cover.SERVICE_OPEN_COVER
         elif position == "custom":
             service = cover.SERVICE_STOP_COVER
-
+    elif instance == f"{light.DOMAIN}.{light.ATTR_EFFECT}":
+        service = light.SERVICE_TURN_ON
+        data[light.ATTR_EFFECT] = mode
```

Comment:
After including the mode identifier, you'll need to extract it in the directive. Also check for supported effects. 
```python
_, effect = mode.split(".")
        if effect in (light.EFFECT_COLORLOOP, light.EFFECT_RANDOM, light.EFFECT_WHITE):
```

Reply:
Same as another comment, different lights have different effects. Exposed via their “effects_list” attribute.

Reply:
Note that @ochlocracy is also saying that you need to split the `mode` to extract the effect.